model/project_ref.go

```diff
@@ -607,6 +609,42 @@ func (p *ProjectRef) GetBatchTime(variant *BuildVariant) int {
 	}
 }
 
+// return the next valid batch time
+func GetActivationTimeWithCron(curTime time.Time, cronBatchTime string) (time.Time, error) {
+	parser := cron.NewParser(cron.Hour | cron.Dom | cron.DowOptional | cron.Descriptor)
+	sched, err := parser.Parse(cronBatchTime)
+	if err != nil {
+		return time.Now(), errors.Wrapf(err, "error parsing cron batchtime '%s'", cronBatchTime)
```

Comment:
It seems odd to return `time.Now()` instead of `0` in the error case. I didn't trace it to the callers, but generally if you return an error you should return a zero value for the other arguments. Otherwise there is some risk that a caller will do something unexpected.

Reply:
Since we don't return an error if GetActivationTime fails, we currently keep the activateAt time as `time.Now()`, which should still create a version. I saw in your comment below you'd rather use the zero time, which would not activate the version. Since this would modify existing behavior, is there any reason we should be cautious? We should also change the behavior in the over arching `GetActivationTime` to match.